### PR TITLE
Splash screen

### PR DESCRIPTION
**Pull request by inyono**

### CODE REVIEW

File: src/components/helpers/Audio.js [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
-      }, this.props.sounds || []);
+      const sounds = filter(identity, this.props.sounds || []);
+
+      this.sounds = map(source => new Audio.Sound(), sounds);
+      this.state = { soundsLoaded: map(() => false, sounds) };
+    }
+
+    componentDidMount() {
+      mapIndexed((sound, i) => {
+        sound.loadAsync(this.props.sounds[i]).then(() => {
```
Comment (Knorrke) [inline src/components/helpers/Audio.js]: Does this work? You filtered `this.props.sounds` to get the `this.sound` list, so I reckon the indices won't match up, or do they?

Comment (inyono) [reply src/components/helpers/Audio.js]: Good catch, I'll change that

**Comment by inyono** [comment]
@Knorrke ready for another review 👀

File: src/components/helpers/Audio.js [hunk 2]
```diff
@@ -1,30 +1,38 @@
...
+      this.sources = filter(identity, this.props.sounds || []);
+      this.sounds = map(_source => new Audio.Sound(), this.sources);
+      this.state = { soundsLoaded: false };
+    }
+
+    componentDidMount() {
+      Promise.all(
+        map(
+          ([sound, source]) => sound.loadAsync(source),
+          xprod(this.sounds, this.sources)
```
Comment (Knorrke) [inline src/components/helpers/Audio.js]: I think this should be `zip`, not `xprod`? Or even more directly judging by the documentation we could use `zipWith` like so:

```
Promise.all(
  zipWith((sound, source) => sound.loadAsync(source), this.sounds, this.sources)
)
).then(() => this.setState({ soundsLoaded: true }));
```

Comment (inyono) [reply src/components/helpers/Audio.js]: 😓 yep

**Comment by inyono** [comment]
continued in #140

**Pull request closed without merging**